sim/paladin/ancient_guardian_pet.go

```diff
@@ -0,0 +1,115 @@
+package paladin
+
+import (
+	"github.com/wowsims/cata/sim/core"
+	"github.com/wowsims/cata/sim/core/proto"
+	"github.com/wowsims/cata/sim/core/stats"
+)
+
+type AncientGuardianPet struct {
+	core.Pet
+
+	paladinOwner *Paladin
+}
+
+func (guardian *AncientGuardianPet) Initialize() {
+}
+
+const PetExpertiseScale = 3.25
+
+func (paladin *Paladin) NewAncientGuardian() *AncientGuardianPet {
+	ancientGuardian := &AncientGuardianPet{
+		Pet: core.NewPet("Ancient Guardian", &paladin.Character, stats.Stats{
+			stats.Stamina: 100,
+		}, func(ownerStats stats.Stats) stats.Stats {
+			return stats.Stats{
+				stats.MeleeHit:  ownerStats[stats.MeleeHit],
+				stats.Expertise: ownerStats[stats.MeleeHit] * PetExpertiseScale,
+
+				stats.MeleeCrit: ownerStats[stats.MeleeCrit],
+			}
+		}, false, true),
+		paladinOwner: paladin,
+	}
+
+	if paladin.Spec == proto.Spec_SpecRetributionPaladin {
+		ancientGuardian.registerRetributionVariant()
+	} else if paladin.Spec == proto.Spec_SpecHolyPaladin {
+		ancientGuardian.registerHolyVariant()
+	}
+
+	ancientGuardian.PseudoStats.DamageTakenMultiplier = 0
+
+	// Guardians are not affected by raid buffs
+	ancientGuardian.PseudoStats.MeleeSpeedMultiplier = 1
+	ancientGuardian.PseudoStats.CastSpeedMultiplier = 1
```

Comment:
I dont think you need this here as guardians already have checks to not have buffs applied to them. Only pets have a very little set of buffs applied to them